core/src/main/java/org/apache/carbondata/core/datamap/DataMapChooser.java

```diff
@@ -282,7 +280,9 @@ private void extractColumnExpression(Expression expression,
       List<Expression> children = expression.getChildren();
       if (children != null && children.size() > 0) {
         for (Expression exp : children) {
-          extractColumnExpression(exp, columnExpressions);
+          if (exp != null && exp.getFilterExpressionType() != ExpressionType.UNKNOWN) {
```

Comment:
I think it's not proper to deal with 'SparkUnknownExpression' specially in framework. You can get the idea from @kevinjmh 's previous PR: The datamap instance know what it wants, so better to let it decide.

Reply:
I'am not sure whether any **other** expression exists like SparkUnknownExpression( which is under EqualToExpression/InExpression and  has a child of ColumnExpression   )